From d92d038a0ce9db8f8eec95620a822afcf960596c Mon Sep 17 00:00:00 2001
From: Prabhat Chand Pandey <prabhat.chand.pandey@intel.com>
Date: Wed, 3 Apr 2019 21:24:55 +0530
Subject: [PATCH] Device : : Intel : Mixin : adding android property

Changes to add android property persist.vendor.sys.usb.adbover, it can
have value dbc or dwc.
Default it will have value dwc.

Tracked-On:
Signed-off-by: Prabhat Chand Pandey <prabhat.chand.pandey@intel.com>
---
 groups/project-celadon/default/init.rc    | 16 +++++++++++++++-
 groups/project-celadon/default/product.mk |  4 ++++
 groups/project-celadon/default/ueventd.rc |  2 +-
 groups/usb-gadget/g_ffs/BoardConfig.mk    |  2 +-
 4 files changed, 21 insertions(+), 3 deletions(-)

diff --git a/groups/project-celadon/default/init.rc b/groups/project-celadon/default/init.rc
index a7d0cdb..1661402 100644
--- a/groups/project-celadon/default/init.rc
+++ b/groups/project-celadon/default/init.rc
@@ -95,7 +95,7 @@ on boot
     write /sys/kernel/debug/pstate_snb/setpoint 75

     # adb over ethernet
-    setprop service.adb.tcp.port 5555
+    # setprop service.adb.tcp.port 5555

 service watchdogd /sbin/watchdogd 10 30
     user root
@@ -118,3 +118,17 @@ on fs
    # Update dm-verity persistent state and set partition.*.verified
    # properties
    verity_update_state
+
+# switch adb over dwc
+on property:persist.vendor.sys.usb.adbover=dwc
+#    write /sys/bus/pci/devices/0000:00:14.0/dbc disable
+    stop adbd
+    start adbd
+
+# switch adb over dbc
+on property:persist.vendor.sys.usb.adbover=dbc
+    write /sys/bus/pci/devices/0000:00:14.0/dbc enable
+    write /sys/bus/pci/devices/0000:00:15.0/dbc enable
+    write /sys/bus/pci/devices/0000:00:39.0/dbc enable
+    stop adbd
+    start adbd
diff --git a/groups/project-celadon/default/product.mk b/groups/project-celadon/default/product.mk
index 55a4494..b9c95c8 100644
--- a/groups/project-celadon/default/product.mk
+++ b/groups/project-celadon/default/product.mk
@@ -102,6 +102,10 @@ PRODUCT_PROPERTY_OVERRIDES += \
 PRODUCT_PROPERTY_OVERRIDES += \
     ro.input.noresample=1

+# set default DBC configuration
+PRODUCT_PROPERTY_OVERRIDES += \
+    persist.vendor.sys.usb.adbover=dwc
+
 # AOSP Packages
 PRODUCT_PACKAGES += \
     Launcher3 \
diff --git a/groups/project-celadon/default/ueventd.rc b/groups/project-celadon/default/ueventd.rc
index ebf9c61..c7ab05b 100644
--- a/groups/project-celadon/default/ueventd.rc
+++ b/groups/project-celadon/default/ueventd.rc
@@ -25,5 +25,5 @@
 /dev/ttyUSB*              0660   radio      radio
 /dev/iio:device*          0660   system     system
 /dev/radio0               0660   bluetooth  audio
-
+/dev/dbc_raw*		  0777	 system	    system
 /sys/devices/pci0000:00/0000:00:02.0/drm/card*/card*/intel_backlight brightness 0644 system system
diff --git a/groups/usb-gadget/g_ffs/BoardConfig.mk b/groups/usb-gadget/g_ffs/BoardConfig.mk
index dbc35c6..b4f9b91 100644
--- a/groups/usb-gadget/g_ffs/BoardConfig.mk
+++ b/groups/usb-gadget/g_ffs/BoardConfig.mk
@@ -1,2 +1,2 @@
 BOARD_SEPOLICY_DIRS += device/intel/project-celadon/sepolicy/usb
-
+BOARD_SEPOLICY_DIRS += $(INTEL_PATH_SEPOLICY)/usb-gadget
--
2.21.0
